operations_on_datastructures/trie_multiple_search.cpp

```diff
@@ -0,0 +1,458 @@
+/**
+ * @file
+ * @brief Trie datastructure with search variants
+ * @details
+ * This provides multiple variants of search functions
+ * on a trie structure utilizing STL. The trie is valid
+ * for only English alphabets.
+ * @author Ghanashyam (https://github.com/g-s-k-zoro)
+ */
+
+#include <cassert>
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <queue>
+#include <vector>

```

Comment:
Please provide a description of what the library/header is for (see the example below).
```c++
#include <iostream>   /// for IO operations
#include <vector>    /// for std::vector
```